composer/utils/object_store/object_store_hparams.py

```diff
@@ -161,20 +161,24 @@ def get_object_store_cls(self) -> Type[ObjectStore]:
         return LibcloudObjectStore
 
     def get_kwargs(self) -> Dict[str, Any]:
-        init_kwargs = {
+
+        provider_kwargs = {}
+        if self.region:
+            provider_kwargs['region'] = self.region
+        if self.host:
+            provider_kwargs['host'] = self.host
+        if self.port:
+            provider_kwargs['port'] = self.port
+        provider_kwargs.update(self.extra_init_kwargs)
+
+        return {
             'provider': self.provider,
             'container': self.container,
-            'provider_kwargs': {},
+            'chunk_size': self.chunk_size,
+            'key_environ': self.key_environ,
+            'secret_environ': self.secret_environ,
```

Comment:
why are these moved out of provider

Reply:
They weren't -- originally those were used inside `initialize_object` to read from envvars if they exist and populate `provider`. That functionality is now in the object store's own `__init__`.

Reply:
Right, so before they were added to the `provider` object but now they are moved outside `provider` into the parent json (if im understanding this correctly). Just making sure this is a conscious choice

Reply:
Ah, let me clarify!

The `provider` object has not been changed. Before/after it is still `{'key': ..., 'secret': ..}`.

Pseudo-code before:
```
provider = {
'key', ...,
'secret',... }

if key_environ in os.environ:
    provider['key'] = # read from envvar

# create object
return ObjectStore( provider..)

```

After:
```
provider = {
'key', ...,
'secret',... }


# create object
return ObjectStore( provider, key_envirion)

class ObjectStore

def __init__(provider, key_envirion):

    # logic now lives here
    if key_environ in os.environ:
        provider['key'] = ...

```